src/transformers/pipelines/document_question_answering.py

```diff
@@ -0,0 +1,461 @@
+                    for cls_index in cls_indices:
+                        p_mask[span_idx][cls_index] = 0
+
+            # For each span, place a bounding box [0,0,0,0] for question and CLS tokens, [1000,1000,1000,1000]
+            # for SEP tokens, and the word's bounding box for words in the original document.
+            if "boxes" not in tokenizer_kwargs:
+                bbox = []
+                for batch_index in range(num_spans):
+                    for i, s, w in zip(
+                        encoding.input_ids[batch_index],
+                        encoding.sequence_ids(batch_index),
+                        encoding.word_ids(batch_index),
+                    ):
+                        if s == 1:
+                            bbox.append(boxes[w])
+                        elif i == self.tokenizer.sep_token_id:
+                            bbox.append([1000] * 4)
+                        else:
+                            bbox.append([0] * 4)
+
+                if self.framework == "tf":
+                    raise ValueError("Unsupported: Tensorflow preprocessing for DocumentQuestionAnsweringPipeline")
+                elif self.framework == "pt":
+                    encoding["bbox"] = torch.tensor([bbox])
+
+            word_ids = [encoding.word_ids(i) for i in range(num_spans)]
+
+        return {
+            **encoding,
+            "p_mask": p_mask,
+            "word_ids": word_ids,
+            "words": words,
+        }
+
+    def _forward(self, model_inputs):
+        p_mask = model_inputs.pop("p_mask", None)
+        word_ids = model_inputs.pop("word_ids", None)
+        words = model_inputs.pop("words", None)
+
+        if self.model_type == ModelType.Donut:
+            model_outputs = self.model.generate(**model_inputs)
+        else:
+            model_outputs = self.model(**model_inputs)
+
+        model_outputs["p_mask"] = p_mask
+        model_outputs["word_ids"] = word_ids
+        model_outputs["words"] = words
+        model_outputs["attention_mask"] = model_inputs.get("attention_mask", None)
+        return model_outputs
+
+    def postprocess(self, model_outputs, top_k=1, **kwargs):
+        if self.model_type == ModelType.Donut:
+            answers = self.postprocess_encoder_decoder(model_outputs)
+        else:
+            answers = self.postprocess_extractive_qa(model_outputs, top_k=top_k, **kwargs)
+
+        answers = sorted(answers, key=lambda x: x["score"], reverse=True)[:top_k]
+        if len(answers) == 1:
+            return answers[0]
+        return answers
+
+    def postprocess_encoder_decoder(self, model_outputs, **kwargs):
+        # postprocess
+        sequence = self.tokenizer.batch_decode(model_outputs.sequences)[0]
+        sequence = sequence.replace(self.tokenizer.eos_token, "").replace(self.tokenizer.pad_token, "")
+        sequence = re.sub(r"<.*?>", "", sequence, count=1).strip()  # remove first task start token
+        ret = {
+            "score": 0.5,  # TODO
```

Comment:
@NielsRogge do you know how to calculate the confidence score from Donut's output?

Reply:
You can't, it's a generative model so it generates tokens one at a time. So the only thing you can have are confidence scores per time step

Reply:
Is there any chance you could help me with this? I'm very unfamiliar with this stuff, so I don't actually know how to do that. I'm fine with landing it as is, btw, which is to mark the score as 0.5, and address that as a follow up if you'd prefer.

Reply:
I don't think we add a score if the model doesn't output them, cc @Narsil